planner/filter/complex.go

```diff
@@ -33,7 +33,7 @@ func isComplex(conditions any, seekRelation bool) bool {
 			if op, ok := k.(*mapper.Operator); ok {
 				switch op.Operation {
 				case request.FilterOpOr, request.FilterOpAnd, request.FilterOpNot:
-					if v, ok := v.([]any); ok && len(v) == 1 {
+					if v, ok := v.([]any); ok && len(v) == 0 {
```

Comment:
It is not ideal to keep this as-is long-term, as it will prevent a possible performance-shortcut once John's fetcher rework gets merged, but for now it is a simple way of excluding the index fetcher filter's lack of support for compound filters..

Reply:
Will do.  I think it is valid to submit an empty and/or clause, but probably not really worth the extra code for that :)

- [ ] Remove len 0 check